accumulo/iterators/src/main/java/org/vertexium/accumulo/iterator/model/EdgeLabels.java

```diff
@@ -0,0 +1,91 @@
+package org.vertexium.accumulo.iterator.model;
+
+import org.vertexium.accumulo.iterator.util.ArrayUtils;
+
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class EdgeLabels {
+    private final List<byte[]> labels;
+
+    public EdgeLabels(List<byte[]> labels) {
+        this.labels = labels;
+    }
+
+    public EdgeLabels() {
+        this(new ArrayList<>());
+    }
+
+    public int add(byte[] bytes) {
+        int result = indexOf(bytes);
+        if (result >= 0) {
+            return result;
+        }
+
+        result = labels.size();
+        labels.add(bytes);
+        return result;
+    }
+
+    public int add(byte[] bytes, int offset, int length) {
+        int result = indexOf(bytes, offset, length);
+        if (result >= 0) {
+            return result;
+        }
+
+        result = labels.size();
+        labels.add(Arrays.copyOfRange(bytes, offset, offset + length));
+        return result;
+    }
+
+    public boolean contains(byte[] bytes) {
+        return indexOf(bytes) >= 0;
+    }
+
+    public int indexOf(byte[] bytes) {
+        int i = 0;
```

Comment:
This method should call the one below with offset 0 and length of `bytes.length`